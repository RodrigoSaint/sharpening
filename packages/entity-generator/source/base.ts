export type FieldType = "string" | "int" | "decimal" | "date";

export interface Field {
  name: string;
  isRequired: boolean;
  type: FieldType;
}

export type EntityAction =
  | "pagination"
  | "list"
  | "detail"
  | "create"
  | "update"
  | "delete";

export interface Entity {
  name: string;
  fieldCollection: Field[];
  actionCollection?: EntityAction[];
  formFieldCollection?: string[];
  displayFieldCollection?: string[];
}
