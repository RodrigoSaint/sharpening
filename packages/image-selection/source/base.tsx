import React, { useState } from "react";
import Modal from "@rodrigosaint/modal";
import ManagedImage from "@rodrigosaint/managed-image";

import ImageSelectionModalContent, {
  ImageSelectionModalContentProps,
} from "./content";

export interface ImageSelectionProps<T>
  extends Omit<ImageSelectionModalContentProps<T>, "close"> {
  renderWithoutImage: () => React.ReactNode;
  getImageUrl: (item: T) => string;
  header: string | React.ReactNode;
  initialValue?: T;
}

export default function ImageSelection<T>(props: ImageSelectionProps<T>) {
  const [src, setSrc] = useState<string>(
    props.initialValue ? props.getImageUrl(props.initialValue) : undefined
  );

  const onSelected = React.useCallback(
    (item: T) => {
      setSrc(props.getImageUrl(item));
      props.onSelected(item);
    },
    [props.onSelected, props.getImageUrl]
  );

  return (
    <>
      <Modal
        renderTrigger={() => (
          <ManagedImage
            src={src}
            renderWithoutImage={props.renderWithoutImage}
            renderOnError={props.renderWithoutImage}
          />
        )}
        renderContent={(close) => (
          <ImageSelectionModalContent
            {...props}
            onSelected={onSelected}
            close={close}
          />
        )}
        header={props.header}
      />
    </>
  );
}
