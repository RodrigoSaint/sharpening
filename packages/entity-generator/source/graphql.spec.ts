import {
  getEntityInput,
  getEntityType,
  getEntityPaginationType,
  getListQuery,
  getPaginationQuery,
  getDetailQuery,
  getCreateMutation,
  getUpdateMutation,
  getDeleteMutation,
  getQuery,
  getMutation,
} from "./graphql";
import { Entity } from "./base";

const entity: Entity = {
  name: "User",
  actionCollection: [
    "pagination",
    "list",
    "detail",
    "create",
    "update",
    "delete",
  ],
  fieldCollection: [
    {
      name: "name",
      isRequired: true,
      type: "string",
    },
    {
      name: "email",
      isRequired: false,
      type: "string",
    },
    {
      name: "age",
      isRequired: false,
      type: "int",
    },
  ],
};

describe("Graphql", () => {
  it("generates entity input", () => {
    expect(getEntityInput(entity)).toMatchSnapshot("getEntityInput");
  });
  it("generates entity type", () => {
    expect(getEntityType(entity)).toMatchSnapshot("getEntityType");
  });
  it("generates entity pagination", () => {
    expect(getEntityPaginationType(entity)).toMatchSnapshot(
      "getEntityPaginationType"
    );
  });
  it("get ListQuery", () => {
    expect(getListQuery(entity)).toMatchSnapshot("ListQuery");
  });
  it("get DetailQuery", () => {
    expect(getDetailQuery(entity)).toMatchSnapshot("DetailQuery");
  });
  it("get PaginationQuery", () => {
    expect(getPaginationQuery(entity)).toMatchSnapshot("PaginationQuery");
  });
  it("get CreateMutation", () => {
    expect(getCreateMutation(entity)).toMatchSnapshot("CreateMutation");
  });
  it("get UpdateMutation", () => {
    expect(getUpdateMutation(entity)).toMatchSnapshot("UpdateMutation");
  });
  it("get DeleteMutation", () => {
    expect(getDeleteMutation(entity)).toMatchSnapshot("DeleteMutation");
  });
  it("get Query", () => {
    expect(getQuery(entity)).toMatchSnapshot("getQuery");
  });
  it("get Mutation", () => {
    expect(getMutation(entity)).toMatchSnapshot("getMutation");
  });
});
