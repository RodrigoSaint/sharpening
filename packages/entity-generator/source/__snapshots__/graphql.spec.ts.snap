// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphql generates entity input: getEntityInput 1`] = `
"input UserInput {
	name: String
	email: String
	age: Int
}"
`;

exports[`Graphql generates entity pagination: getEntityPaginationType 1`] = `
"type UserPagination {
	collection: [User]
	currentPage: Int
	pageCount: Int
}"
`;

exports[`Graphql generates entity type: getEntityType 1`] = `
"type User {
	name: String
	email: String
	age: Int
}"
`;

exports[`Graphql get CreateMutation: CreateMutation 1`] = `"createUser(user: UserInput!): User!"`;

exports[`Graphql get DeleteMutation: DeleteMutation 1`] = `"deleteUser(userId: String!): Boolean!"`;

exports[`Graphql get DetailQuery: DetailQuery 1`] = `"user(userId: String!): User!"`;

exports[`Graphql get ListQuery: ListQuery 1`] = `"userCollection: [User]!"`;

exports[`Graphql get Mutation: getMutation 1`] = `
"type Mutation {
	createUser(user: UserInput!): User!
	updateUser(user: UserInput!): User!
	deleteUser(userId: String!): Boolean!
}"
`;

exports[`Graphql get PaginationQuery: PaginationQuery 1`] = `"userPagination(page: Int!): UserPagination!"`;

exports[`Graphql get Query: getQuery 1`] = `
"type Query {
	userPagination(page: Int!): UserPagination!
	userCollection: [User]!
	user(userId: String!): User!
}"
`;

exports[`Graphql get UpdateMutation: UpdateMutation 1`] = `"updateUser(user: UserInput!): User!"`;
